synapse/api/constants.py

```diff
@@ -118,6 +119,7 @@ class RoomVersions(object):
     RoomVersions.V2,
     RoomVersions.VDH_TEST,
     RoomVersions.STATE_V2_TEST,
+    RoomVersions.V3,
```

Comment:
We can't add V3 here yet, because doing so will allow us to accept joins/creates for V3 rooms, but won't actually implement the things that make V3 V3.